src/main/scala/org/constellation/p2p/Download.scala

```diff
@@ -426,12 +412,23 @@ class DownloadProcess[F[_]: Concurrent: Timer: Clock](
       .flatMap(_ => snapshotsProcessor.processSnapshots(hashes))
       .flatTap(_ => dao.metrics.updateMetricAsync("downloadSecondPassComplete", "true"))
 
-  private def finishDownload(snapshot: SnapshotInfo): F[Unit] =
+  private def removeSnapshotInfoFromDisk(snapshotHeight: Long,
+                                         overWritePath: String = dao.snapshotInfoPath.pathAsString) = {
+    val snapshotInfoAtHeight = dao.snapshotInfoPath.collectChildren(_.isDirectory)
+      .filter { file =>
```

Comment:
We can have only one `SnapshotInfo` at specific height so it shouldn't return `Seq` but just `Option`.

Just use `find` instead of `.filter(...).toList` because we always get one item or nothing.

Reply:
Totally, but glad you mentioned so be warned: I saw that multiple snapshotInfos at the same height were being written to disk at one point and I wrote it this way to ensure deletion. I believe I have fixed the underlying issues with snapshotInfos, and the rest are with snapshots on disk. However, double check when rewriting as .find